src/XCCDF/xccdf_session.c

```diff
@@ -524,13 +524,43 @@ static int _acquire_xccdf_checklist_from_tailoring(struct xccdf_session* session
 		xmlFree(tailoring_xmlDoc);
 		return 1;
 	}
-	char *benchmark_id = strchr(benchmark_ref, '#') + 1;
 
-	struct oscap_source *xccdf_source = ds_sds_session_select_checklist(ds_sds_session,
-			NULL, benchmark_id, NULL);
+	struct oscap_source *xccdf_source = NULL;
+	if (oscap_str_startswith(benchmark_ref, "file:")) {
+		char* sep = strchr(benchmark_ref, '#');
+		const char *filename = benchmark_ref + strlen("file:");
+		const char *component_ref = NULL;
+		if (sep != NULL) {
+			component_ref = sep + 1;
+			*sep = '\0';
+		}
+
+		struct oscap_source *external_file = oscap_source_new_from_file(filename);
+		if (oscap_source_get_scap_type(external_file) == OSCAP_DOCUMENT_SDS) {
```

Comment:
It is good to check, could you please add this condition to sds.c where can be found similar code?